File: src/main/kotlin/moe/kurenai/tdlight/model/keyboard/ReplyKeyboardRemove.java
package moe.kurenai.tdlight.model.keyboard;

import com.fasterxml.jackson.annotation.JsonProperty;
import moe.kurenai.tdlight.model.message.ReplyMarkup;

import java.util.Optional;

public record ReplyKeyboardRemove(
        @JsonProperty("remove_keyboard") Boolean removeKeyboard,
        @JsonProperty("selective") Optional<Boolean> selective
) implements ReplyMarkup {
}

File: src/main/kotlin/moe/kurenai/tdlight/model/keyboard/ReplyKeyboardRemove.kt
package moe.kurenai.tdlight.model.keyboard

import com.fasterxml.jackson.annotation.JsonProperty
import moe.kurenai.tdlight.annotation.NoArg
import moe.kurenai.tdlight.model.ReplyMarkup

@NoArg
data class ReplyKeyboardRemove(

    @JsonProperty("remove_keyboard")
    val removeKeyboard: Boolean,

    @JsonProperty("selective")
    val selective: Boolean? = null
) : ReplyMarkup {
}